package.json

```diff
@@ -129,6 +157,10 @@
     ]
   },
   "xo": {
-    "prettier": true
+    "prettier": true,
+    "rules": {
+      "max-params": "off",
+      "no-undef": "off"
```

Comment:
Can this one be inlined into the places that needs it? Or, can you add `global` comments to enable what you are expected?

Reply:
I have mixed feelings about this. On it’s own, I believe this is an extremely useful rule. However, TypeScript also checks the same thing, so I think it loses its value.

Reply:
Thing for me is that this lint message is a symptom and the actual problem is an unconfigured environment: https://eslint.org/docs/latest/user-guide/configuring/language-options#specifying-environments. Environments can affect more things though (especially with xo and presets it might do much more than ESLint by default, which is mostly globals and `ecmaVersion`).

AFAIK TS shows this in some editors, but compilation doesn’t crash on it, so it wouldn’t work in a CI?

Reply:
> Thing for me is that this lint message is a symptom and the actual problem is an unconfigured environment

I agree with that. After configuring `env`, the only false positive is a custom variable injected by the Monaco editor TypeScript worker. Cross-referencing https://github.com/microsoft/monaco-editor/pull/3488.

> AFAIK TS shows this in some editors, but compilation doesn’t crash on it, so it wouldn’t work in a CI?

If TypeScript shows this in the editor, and `tsc` is run in CI, it will most definitely be caught.